diff --git a/src/runtime/client/remote-functions/query/index.js b/src/runtime/client/remote-functions/query/index.js
index 3ea8ad479..56e411430 100644
--- a/src/runtime/client/remote-functions/query/index.js
+++ b/src/runtime/client/remote-functions/query/index.js
@@ -30,7 +30,10 @@ export function query(id) {
 
 			if (result.redirect) {
 				await goto(result.redirect);
+				return;
 			}
+
+			return result._;
 		});
 	};
 
